python/pyproject.toml

```diff
@@ -40,6 +40,7 @@ duckduckgo-search = "^7.3.2"
 json-repair = "^0.39.0"
 wikipedia-api = "^0.8.1"
 async-generator = "^1.10"
+boto3 = "^1.37.5"
```

Comment:
Is this needed?

Reply:
yes - at least to use bedrock.
Alternatively we do not code as a dependency, and the backend will fail if the user hasn't installed. I opted for inclusion.

Reply:
See [LiteLLM docs](https://docs.litellm.ai/docs/providers/bedrock) and statement _LiteLLM requires boto3 to be installed on your system for Bedrock requests_

Reply:
In that case, we should let the user install it if needed. For our purposes I would add `boto3` to dev dependencies.

Reply:
done (moved to dev)